### ISSUE TITLE
AWS Toolkit has the slowest activation time among all my extensions

### ISSUE THREAD

**Issue by tigerinus** [aws/aws-toolkit-vscode#1918]
**Describe the bug**

AWS Toolkit has the slowest activation time among all my 30+ extensions

**To Reproduce**

1. Boot VSCode (see version info below)
2. Check extension activation time

**Expected behavior**

Should be fast to activate.

**Screenshots**

![image](https://user-images.githubusercontent.com/7172560/128413172-4a269f9b-58d5-41b9-abfd-d35d46a2bd9f.png)

**Desktop (please complete the following information):**

Version: 1.59.0 (system setup)
Commit: 379476f0e13988d90fab105c5c19e7abc8b1dea8
Date: 2021-08-04T23:13:12.822Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Windows_NT x64 10.0.19043

**Additional context**

**Comment by JadenSimon** [comment]
Looking into it. If this is related to our activation code then we should be able to get some speed-ups somewhere.

**Comment by JadenSimon** [comment]
Update:

There are currently no easy solutions for decreasing start-up time, but we do have a few ideas. The first major boost should be when we finish migrating to v3 of the JS AWS SDK. That'll reduce the size of the main extension script substantially. Then we can work towards optimizing the extension build configuration (e.g. tree-shaking) which again will reduce extension size. 

Some other gains could be had through little micro-optimizations but those mostly affect what happens after the extension is loaded. For example, VS Code currently reports that the toolkit has been activated even though some of the command registration happens asynchronously in the background. We would want to minimize the time between extension load and actual command registration.

**Comment by CareyJWilliams** [comment]
Hi @JadenSimon are there any updates on this?

**Comment by justinmk3** [comment]
Yes. We made startup time improvements recently, and plan to do more. See also https://github.com/aws/aws-toolkit-vscode/issues/3216

**Comment by justinmk3** [comment]
Further performance improvements: in AWS Toolkit 1.86, all directories mentioned by the standard vscode settings `files.exclude`, `search.exclude`, and `files.watcherExclude` are ignored by the "Scanning CloudFormation templates" feature, which should result in faster startup on big workspaces, and provides a flexible workaround for any big directories not already ignored.
